opm/parser/eclipse/EclipseState/Grid/GridProperties.hpp

```diff
@@ -58,13 +58,22 @@ namespace Opm {
     void setKeywordBox( const DeckRecord& deckRecord,
                         BoxManager& boxManager);
 
-
     class Eclipse3DProperties;
 
+    template <typename T> class GridPropertyIterator;
+
+    template <typename T>
+    using storageType = typename std::map<std::string , std::shared_ptr<GridProperty<T> > >;
+
+    template <typename T>
+    using  storageIteratorType = typename std::map<std::string , std::shared_ptr<GridProperty<T> > >::const_iterator;
```

Comment:
Why don't you leverage your own aliases?

`template< typename T >` using const_iterator = storage< T >::const_iterator`